spec/index.bs

```diff
@@ -873,6 +922,97 @@ interface GPUBufferUsage {
 
 ## Buffer Mapping ## {#buffer-mapping}
 
+An application can request to map a {{GPUBuffer}} to get its mapping which is
+an {{ArrayBuffer}} representing the {{GPUBuffer}}'s allocation. Mappings are
+requested asynchronously so that the user agent can ensure the GPU is finished
+using the {{GPUBuffer}} before the application gets its mapping. Mappings can
+be requested for reading with {{GPUBuffer/mapReadAsync}} or writing with
+{{GPUBuffer/mapWriteAsync}}. A mapped {{GPUBuffer}} cannot be used by the GPU
+and must be unmapped using {{GPUBuffer/unmap}} or {{GPUBuffer/destroy}} before
+it can be used by the GPU again.
+
+<!-- TODO(kangz) add client-side validation that a mapped buffer can only be destroyed on the worker on which it was mapped -->
+
+### {{GPUBuffer/mapReadAsync|GPUDevice.mapReadAsync}} ### {#GPUBuffer-mapReadAsync}
+
+<dl dfn-type="method" dfn-for="GPUBuffer">
+    : <dfn>mapReadAsync()</dfn>
+    ::
+        <div algorithm="GPUBuffer.mapReadAsync()">
+            <!-- TODO(kangz) handle error buffers once we have a description of the error monad -->
+            1. If the {{[[usage]]}} slot of |this| doesn't contain the {{GPUBufferUsage/MAP_READ}} bit:
+
+                1. Record a validation error on the current scope.
+                1. Return a new promise that rejects. <!-- TODO(kangz) specify on that it happens on the device timeline-->
+
+            1. Let |p| be a new {{Promise}}.
+            1. Set the {{[[mapping_promise]]}} slot of |this| to |p|.
+            1. Set the {{[[state]]}} slot of |this| to [=buffer state/mapped for reading=]
+            1. Enqueue an operation on the [=Queue timeline=] that will execute the following:
```

Comment:
Usually, spec use the wording below:
> Return p and run the following steps in parallel:

So you may want to say something like
> Return p and run the following steps in parallel:
> - Let |m| be a new {{ArrayBuffer}} of size the {{[[size]]}} of |this|.
> - Set the content of |m| to the content of |this|'s allocation.
> - Resolve |p| with |m| if it is not already rejected.

Reply:
fwiw I like the original wording, which uses our "queue timeline" to be more precise than the usual "in parallel".